### PR TITLE
Dead raptors no longer get happy when petted or groomed

### PR DESCRIPTION
**Pull request by Profakos**
## About The Pull Request

I have noticed that when I petted a raptor corpse I have found, it started spinning and hearts appeared above it. Apparently, the same happens if you wash them. This PR makes sure that dead raptors stop spinning in their graves.

I decided to not to disable the happiness component's processing on death, as the idea that the raptor's ghost looms over you, patiently waiting to be revived is funny, and also encourages their revival to be a priority.

This PR also removes a duplicate animal petting proc that has not actually been hooked into anything.

## Why It's Good For The Game

Dead animals should consistently act as dead animals.

## Changelog

:cl:
fix: Fixes dead raptors getting excited when petted or groomed
/:cl:

### CODE REVIEW

**Comment by Dawnseer** [comment]
But I want my raptor to be happy in the afterlife...

**Comment by the-orange-cow** [comment]
B-b-but t-the QM told me he was just sleeping!!

File: code/datums/components/happiness.dm [hunk 1]
```diff
@@ -56,7 +56,7 @@
 	ADD_TRAIT(parent, TRAIT_MOB_RELAY_HAPPINESS, REF(src))
 
 /datum/component/happiness/UnregisterFromParent()
-	UnregisterSignal(parent, list(COMSIG_HOSTILE_PRE_ATTACKINGTARGET, COMSIG_COMPONENT_CLEAN_ACT, COMSIG_MOB_ATE))
+	UnregisterSignal(parent, list(COMSIG_HOSTILE_PRE_ATTACKINGTARGET, COMSIG_COMPONENT_CLEAN_ACT, COMSIG_MOB_ATE, COMSIG_LIVING_DEATH, COMSIG_LIVING_REVIVE))
```
Comment (Jacquerel) [inline code/datums/components/happiness.dm:59]: Is this an artifact of an earlier version of your change? It doesn't look like those signals are registered.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/datums/components/happiness.dm (hunk 1 above)
Comment (Profakos) [reply code/datums/components/happiness.dm:59]: You are right, I tried to make their happiness pause on death and resume on life, but I decided against it, and did not fully clean it up. I will fix this today.

**Pull request merged**